### ISSUE TITLE
Suggestions list is empty when initially triggering with Ctrl+Space, need to "press the key combo twice"

### ISSUE THREAD

**Issue by dmalyuta** [microsoft/pylance-release#2112]
Originally posted [here](https://github.com/microsoft/vscode/issues/137827).

## Environment data

-   Language Server version: v2021.11.2
-   OS and version: Ubuntu 16.04.7 LTS
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.7.4

## Expected behaviour

Using the following user settings.json in order to not get completions popping up unless I press Ctrl+Space to ask for them manually:
```
    "editor.quickSuggestions": { "other": false, "comments": false, "strings": false },
    "editor.quickSuggestionsDelay": 1000,
    "editor.suggestOnTriggerCharacters": false,
    "editor.acceptSuggestionOnCommitCharacter": false,
    "editor.wordBasedSuggestions": false,
    "editor.parameterHints.enabled": false,
    "editor.suggest.snippetsPreventQuickSuggestions": false,
    "editor.inlineSuggest.enabled": false,
    "editor.snippetSuggestions": "top",
```

Expect the suggestions list to be populated with available completions whenever I press Ctrl+Space (`editor.action.triggerSuggest`).

## Actual behaviour

Here's what happens instead:

1. Open a Python file with extension ms-python.python and ms-python.vscode-pylance
2. Type `im` in the Python file
3. Press Ctrl+Space to trigger suggestions list (`editor.action.triggerSuggest`). It's empty.
4. Press Esc and do Ctrl+Space again. Now the suggestions list is populated (with `import` up top, for example)!

Basically, the suggestions list only gets populated on the _second_ time that I press Ctrl+Space to trigger completions (exact key sequence: Ctrl+Space then Esc then Ctrl+Space). After a minute or so, the list becomes empty again and I need to do Ctrl+Space twice again.

**Comment by judej** [comment]
Thanks for the report. Have you tried turning on indexing? `python.analysis.indexing: True `

**Comment by dmalyuta** [comment]
Thanks for the suggestion, here's what I put into my VS code user settings JSON:
```
    /// Language: Python
    "jupyter.interactiveWindowMode": "perFile",
    "jupyter.askForKernelRestart": false,
    "pylance.insidersChannel": "daily",
    // "python.analysis.logLevel": "Trace",
    "python.analysis.indexing": true,
```

However the line `"python.analysis.indexing": true,` is faded out and hovering over it says "Unknown Configuration Setting". 

In any case, I tried with that line and no change in behavior. Overall, it just seems like the suggestions widget is "too quick" and pops up with zero completions while python LSP is still analyzing completions in the background. The correct behavior would be for the widget to asynchronously wait for the LSP server to return the full completion suggestions list - no matter if it ends up having something or returns empty.

**Comment by heejaechang** [comment]
I don't think the issue is related to indexing. so the behavior you see won't be affected by indexing option. that being said, I can't repro the issue once we are initialized. did you wait until LS is fully initialized before invoking completion? we won't even get completion request from editor until we are initialized. you can see whether we are initialized from "Python Language Server" in "output" window. it should say "Background analsysis started"

> The correct behavior would be for the widget to asynchronously wait for the LSP server to return the full completion suggestions list

we already does that once we are initialized. but if we are in the middle of initializing, no events will come to us until they tell editors that we are ready.

**Comment by heejaechang** [comment]
closing it as by design.
